### ISSUE TITLE
Support for searching by instance

### ISSUE THREAD

**Issue by Porrumentzio** [tootsuite/mastodon#15097]
### Pitch

Currently there's no aparently way to search public accounts on an instance, but they're searchable with the whole username.

It would be great if searching for the instance, results would show accounts having that instance address.
This could also apply for any AP instance.

### Motivation
I frustrate myself by not being able to search accounts on interesting instances, when I would like to. As some instance are theme or language specific, this is a good way to discover accounts.

**Comment by sgornick** [comment]
To understand correctly what you are asking, are you saying you would like a way to enter an instance and see all the user accounts on that instance?  

With a specific Mastodon instance in which I have an account, BitcoinHackers.org, it appears I can do just that.  In the search (e.g., using the /web/search path), I can enter an instance, e.g., Mastodon.social, and the results under the People section include users on that instance.  Clicking the Load More button lets me see additional users.

But this does not appear to be possible from every instance.  For example, I also have an account on view Mstdn.io.  From there, I can enter into the search the instance mastodon.social, and it appears to top out with a maximum of three results that are users on the mastodon.social instance (plus some other matches from other instances in which mastodon.social is matched based on the user's Display Name).   Additionally, from Mstdn.io, there are some instances it yields no results for -- such as BitcoinHackers.org, which I assume is due to Mstdn.io following [FediBlock's Block Recommendation List](https://fediblock.org/blocklist) and thus rejects that instance.

So it appears that the ability to, from your own Mastodon instance, use search to discover users of other Mastodon instances is limited by the way your instance is configured.  The thing is, I don't know if the search in BitcoinHackers.org instance (which includes in the search results users from other instances without a limit as to how many) is the default configuration, or if the way search in Mstdn.io (which includes a maximum of three users from other instances) occurs is the default configuration.

There are workarounds:

- https://search.noc.social
- An instance's /public page shows Toots, and from that you might follow users in which you are interested.  For https://mastodon.social/public  (I also see https://mastodon.social/explore, which is similar, but only Mastodon.social has this, it appears). 
  
Also, there is a related Issue #9529 that you might wish to read.

**Comment by Porrumentzio** [comment]
Yeah, what I would like is to be able to search all the users from an instance, as you said, but from inside my own instance, or even Tusky or any other app (which does not support this because Mastodon itself doesn't, AFAIK)

That other proposal is interesting, could have something to see
